azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -558,6 +563,43 @@ def to_json(self):
         return json.dumps({"Options": self._options, "Parameters": self._parameters}, default=str)
 
 
+class ExecuteRequestParams:

```

Comment:
I'm not sure why you need a class here

Reply:
What do you think would be good instead?

Have a function that returns a 3 tuple?
Or split it into multiple functions that you have to call in the correct order (since timeout is dependent on the headers)?

I do agree it's a bit ugly, but I don't have a better idea..